test/screenshotter/test.html

```diff
@@ -32,31 +32,40 @@
     <span id="math"></span>
     <span id="post"></span>
     <script type="text/javascript">
-      var query = {};
-      var re = /(?:^\?|&)([^&=]+)(?:=([^&]+))?/g;
-      var match;
-      while (match = re.exec(window.location.search)) {
-          query[match[1]] = decodeURIComponent(match[2]);
-      }
-      var mathNode = document.getElementById("math");
+      function handle_search_string(search, callback) {
+        var query = {};
+        var re = /(?:^\?|&)([^&=]+)(?:=([^&]+))?/g;
+        var match;
+        while (match = re.exec(search)) {
+            query[match[1]] = decodeURIComponent(match[2]);
+        }
+        var mathNode = document.getElementById("math");
 
-      var settings = {
-          displayMode: !!query["display"],
-          throwOnError: !query["noThrow"]
-      };
-      if (query["errorColor"]) {
-          settings.errorColor = query["errorColor"];
-      }
-      var macros = {};
-      var macroRegex = /(?:^\?|&)(?:\\|%5[Cc])([A-Za-z]+)=([^&]*)/g;
-      while ((match = macroRegex.exec(window.location.search)) !== null) {
-          settings.macros = macros;
-          macros["\\" + match[1]] = decodeURIComponent(match[2]);
-      }
+        var settings = {
+            displayMode: !!query["display"],
+            throwOnError: !query["noThrow"]
+        };
+        if (query["errorColor"]) {
+            settings.errorColor = query["errorColor"];
+        }
+        var macros = {};
+        var macroRegex = /(?:^\?|&)(?:\\|%5[Cc])([A-Za-z]+)=([^&]*)/g;
+        while ((match = macroRegex.exec(search)) !== null) {
+            settings.macros = macros;
+            macros["\\" + match[1]] = decodeURIComponent(match[2]);
+        }
 
-      katex.render(query["tex"], mathNode, settings);
-      document.getElementById("pre").innerHTML = query["pre"] || "";
-      document.getElementById("post").innerHTML = query["post"] || "";
+        katex.render(query["tex"], mathNode, settings);
+        document.getElementById("pre").innerHTML = query["pre"] || "";
+        document.getElementById("post").innerHTML = query["post"] || "";
+
+        if (callback && document.fonts && document.fonts.ready) {
+          document.fonts.ready.then(callback);
```

Comment:
I didn't realize there was a `document.fonts.ready` was a thing.  I've used webfontloader in the past, but this is better for this use.